File: Solutions/[01]_2.java
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int n, m;
		int[] data = new int[100];
		int result = 0;
		
		n = sc.nextInt();
		m = sc.nextInt();
		for (int i = 0; i < n; i++) {
			data[i] = sc.nextInt();
		}

		for (int i = 0; i < n; i++) {
			for (int j = i + 1; j < n; j++) {
				for (int k = j + 1; k < n; k++) {
					int sum = data[i] + data[j] + data[k];
					if (sum <= m) result = Math.max(result, sum);
				}
			}
		}
		System.out.println(result);
	}

}

File: Solutions/[01]_2.cpp
#include <bits/stdc++.h>

using namespace std;

int n, m;
int data[100];
int result = 0;
	
int main() {
	cin >> n >> m;
	for (int i = 0; i < n; i++) {
		cin >> data[i];
	}
	for (int i = 0; i < n; i++) {
		for (int j = i + 1; j < n; j++) {
			for (int k = j + 1; k < n; k++) {
				int sum = data[i] + data[j] + data[k];
				if (sum <= m) result = max(result, sum);
			}
		}
	}
	cout << result;
}
